add PUT Request and add EditMember Form

Fabi911 opened the pull request.

Fabi911 committed:
Add member update functionality

Implemented updateMember method in MemberService and corresponding endpoint in MemberController. Added unit tests in MemberServiceTest and integration tests in MemberControllerTest to ensure functionality.

Fabi911 committed:
Add EditMember Form and Update Member Details Flow

Introduced a form for editing member details and integrated it into the MemberDetail page. Enhanced MembersOverview with loading skeletons and new icons. Adjusted modal styling and added relevant dependencies in package.json.

rinaehyun commented on frontend/src/Pages/MemberDetail.tsx:
```diff
@@ -27,13 +33,45 @@ export default function MemberDetail() {
+                <span>Tel: {phoneNumber}</span>
+                <span>E-Mail: {email}</span>
+                <span>Geburtstag: {new Date(birthday).toLocaleDateString()}</span>
+<EditButton onClick={() => setModal(true)}><EditNoteIcon fontSize="large"/> </EditButton>
```
Align this line please

rinaehyun commented:
Generally, looks good to me.
Can I ask you one thing? If the user wants to update an Item, which does not exist in DB, then what happens? There would be an exception handling or..?

Fabi911 commented:
> Generally, looks good to me. Can I ask you one thing? If the user wants to update an Item, which does not exist in DB, then what happens? There would be an exception handling or..?

at the moment this function is only possible in the detail view of a member, the id is already checked.
But if it happens that the id is not available, the member is saved as a new member. See @MemberService.java line 36.

Fabi911 committed:
Refactor alignment for readability

Improve readability by adjusting the alignment of the EditButton component and its wrapping Modal. Updated formatting to ensure code consistency and enhance maintainability.

rinaehyun commented:
Okay :) then everything looks good to me :)

rinaehyun reviewed (approved):
good :)

rinaehyun reviewed (approved):
Approved :)
